component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -561,13 +631,188 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
+        if (newAccessResponseCode == HttpConstants.SUCCESS_CODE) {
+            String newAccessToken = "Bearer " + newAccessTokenArray.get(1);
+            accessTokenCache.setAccessToken(encodedRefreshAuth, newAccessToken);
+            if (!newAccessTokenArray.get(2).equals(HttpConstants.EMPTY_STRING)) {
+                accessTokenCache.setRefreshtoken(encodedRefreshAuth, newAccessTokenArray.get(2));
+            }
+            for (Header header : headersList) {
+                if (header.getName().equals(HttpConstants.AUTHORIZATION_HEADER)) {
+                    header.setValue(newAccessToken);
+                }
+            }
+            //send a request to API with a new access token
+            int response = sendRequest(payload, dynamicOptions, headersList);
+            if (response == HttpConstants.SUCCESS_CODE) {
+                log.info("Request send successfully.");
+            } else if (response == HttpConstants.AUTHENTICATION_FAIL_CODE) {
+                log.error(response + ": Authentication Failure. Please apply valid authorization.");
+            } else if (response == HttpConstants.INTERNAL_SERVER_FAIL_CODE) {
+                log.error(response + ": Internal server error.");
+            } else {
+                log.error(response + ": Can not obtain new access token.");
+            }
+
+        } else if (newAccessResponseCode == HttpConstants.AUTHENTICATION_FAIL_CODE) {
+            log.error(newAccessResponseCode + ": Authentication Failure. Please apply valid Consumer key, " +
+                    "Consumer secret for generate new access token. ");
+        } else {
+            log.error(newAccessResponseCode + ": Can not obtain new access token.");
+        }
+
+    }
+
+    public ArrayList<String> getAccessToken(DynamicOptions dynamicOptions, String encodedRefreshAuth) {
+        HttpsClient httpsClient = new HttpsClient();
+        ArrayList<String> newAccessTokenArray;
+        if (!HttpConstants.EMPTY_STRING.equals(oauthUsername) &&
+                !HttpConstants.EMPTY_STRING.equals(oauthUserPassword)) {
+            newAccessTokenArray = httpsClient.getPasswordGrandAccessToken(publisherURL, clientStoreFile,
+                    clientStorePass, oauthUsername, oauthUserPassword, encodedRefreshAuth);
+        } else if (!HttpConstants.EMPTY_STRING.equals(refreshToken.getValue(dynamicOptions)) ||
+                !availableRefreshToken.equals(HttpConstants.EMPTY_STRING)) {
+            newAccessTokenArray = httpsClient.getRefreshGrandAccessToken(publisherURL, clientStoreFile,
+                    clientStorePass, encodedRefreshAuth, refreshToken.getValue(dynamicOptions));
+        } else {
+            newAccessTokenArray = httpsClient.getClientGrandAccessToken(publisherURL, clientStoreFile,
+                    clientStorePass, encodedRefreshAuth);
+        }
+        return newAccessTokenArray;
+    }
+
+    public void checkAccessToken(String encodedRefreshAuth, DynamicOptions dynamicOptions,
```

Comment:
Shall we call this method `setAccessToken` instead because when I read `checkAccessToken` it gives me a feeling like you are checking whether access token is there or not and this method should return a boolean. WDYT?